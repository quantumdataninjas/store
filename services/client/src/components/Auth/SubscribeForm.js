import React, { Component } from 'react'
import axios from 'axios'

class SubscribeForm extends Component {
  constructor() {
    super()
    this.state = {
      value: ''
    }
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(event) {
    this.setState({value: event.target.value});
  }

  handleSubmit(event) {
    axios({
      method: 'post',
      url: `${process.env.REACT_APP_USERS_SERVICE_URL}/users/subscribe`,
      data: {
        email: this.state.value
      },
      headers: {
        'Content-Type': 'application/json'
      }
    })
    .then((res) => { console.log(res); })
    .catch((err) => { console.log(err); });
    event.preventDefault();
  }

  render() {
    return (
      <div className="container">
        <form onSubmit={this.handleSubmit}>
          <label>
            Email:
            <input type="text" name="email" value={this.state.value} onChange={this.handleChange} />
          </label>
          <input type="submit" value="submit" />
        </form>
      </div>
    )
  }
}

export default SubscribeForm;
