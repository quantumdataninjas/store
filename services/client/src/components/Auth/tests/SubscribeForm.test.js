import React from 'react'
import { mount, shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import SubscribeForm from '../SubscribeForm'
import { TextField, Button } from '@material-ui/core'



describe('SubscribeForm Component', () => {

    it('renders', () => {
        const wrapper = mount(<SubscribeForm />)
        const form = wrapper.find('form')
        expect(form.find('input')).toHaveLength(1)
        expect(form.find(TextField)).toHaveLength(1)
        expect(form.find(TextField).get(0).props.name).toBe('email')
        expect(form.find(Button).get(0).props.type).toBe('submit')
    })

    it('renders a snapshot', () => {
        const tree = renderer.create(<SubscribeForm />)
        expect(tree).toMatchSnapshot()
    })
})