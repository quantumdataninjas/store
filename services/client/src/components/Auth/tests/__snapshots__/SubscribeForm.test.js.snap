// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubscribeForm Component renders a snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 SubscribeForm-container-1 MuiPaper-rounded"
>
  <form
    autoComplete="on"
    className="SubscribeForm-form-2"
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root MuiTextField-root SubscribeForm-textField-3 MuiFormControl-marginNormal"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
        data-shrink={false}
        htmlFor="email-field"
      >
        Email
        <span
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
          â€‰
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          id="email-field"
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="email"
        />
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root SubscribeForm-button-5 MuiButton-text"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
        Subscribe
      </span>
    </button>
  </form>
</div>
`;
